libmscore/undo.h

```diff
@@ -898,6 +899,24 @@ class ChangeNoteEvents : public UndoCommand {
       UNDO_NAME("ChangeNoteEvents")
       };
 
+//---------------------------------------------------------
+//   ChangeNoteEventList
+//---------------------------------------------------------
+
+class ChangeNoteEventList : public UndoCommand {
+      Ms::Note*      note;
+      NoteEventList* oldEvents;
```

Comment:
I did this for clarity (of which I could find none in the undo code.)

What I see when debugging is that flip() gets called when you call the score->undo() and it actually applies the new data when the undo is registered. 

This quite surprised me because it put back the data I just applied thereby undo'ing my do! I thought I should at least be explicit about it.

If there's a better way please suggest an actual implementation. This is where I ended up so I actually put comments in to describe the process--as confusing as it was. 

I will note that it also works really well!

-Dale

Reply:
If `flip()` is undoing your do, then you are certainly doing something wrong. `flip()` should be what does the do in the first place.

Reply:
>flip() should be what does the do in the first place.

Exactly my experience. But I had to discover that by debugging. 

When I added the Change* thing using ->undo() my expectation was I was registering the object used to UNDO the operation. It turns out that this is not the case. It's also responsible for the actual DO. For me this was not intuitive in the least and led to my opening a topic on the forum about undo to help clarify my confusion.

https://musescore.org/en/node/291519#comment-930533

This didn't get me any closer to a design I could look at so the code is my guru.

-Dale

Reply:
I cleaned up the undo for PlayEventLists.